src/shamalgs/src/primitives/reduction.cpp

```
@@ -28,40 +28,58 @@
-        };
-
-        /// USM group reduction, 256-wide work groups
-        struct GroupReduction256 {
-            static constexpr std::string_view variant_type_name = "group_reduction256";
-        };
+template<>
+struct shamalgs::ImplVariantParams<shamalgs::primitives::impl::GroupReduction> {
+    static nlohmann::json to_json(const shamalgs::primitives::impl::GroupReduction &p) {
+        return {{"group_size", p.group_size}};
+    }
+    static shamalgs::primitives::impl::GroupReduction from_json(const nlohmann::json &j) {
```

Cppcheck (2.21.0)

[style] The function 'from_json' is never used.

(unusedFunction)